Arduino-Bling/RobotBling/RobotBling.ino

```diff
@@ -0,0 +1,209 @@
+/*
+ * Spartronics Arduino Robot Bling Integration
+ *
+ * Arduino code for animation support:
+ * - uses rgbColor struct to break color into its components for easy reference
+ * - listens on serial port for new animation commands
+ * - ALL animations require call to `delay()` tp listen for interruptions
+ *
+ * Important:
+ * - Expected commands: ascii printable characters 0 and above
+ * - For development, start Arduino's serial port to test commands and animations
+ *
+ */
+
+/*
+ * Used by the magic _delay() function for running animations
+ */
+#include <setjmp.h>
+
+/*
+ * Setup Adafruit NeoMatrix
+ * - Important: validate PIN connection and number of LEDs
+ * - Note: this NeoMatrix does not have 'w' component
+ */
+#include <Adafruit_NeoPixel.h>
+
+// Configuration for our LED strip and Arduino
+// TODO: Update PIN and NUM_LEDS for development and later robot deployment
+#define NUM_LEDS 60
+#define PIN 4
+
+// allocate our pixel memory, set interface to match our hardware
+Adafruit_NeoPixel pixels = Adafruit_NeoPixel(NUM_LEDS, PIN, NEO_GRB + NEO_KHZ800);
+
+// save the desired system wide brightness value -- 0:255 --> level 32 == 1/8th the brightness
+uint8_t defaultSystemBrightness = 32;
+
+/*
+ * struct rgbColor --
+ * Animations require easy access to color information: both full color, or its rgb parts
+ * rgbColor struct supports this via the magic of 'union'
+ *  - `union` allows storing different data types in the same memory location
+ *  - different rgbColor constructors enables easy parsing of data
+ * Reminder: we don't have 'w' component in our neopixels -- rgb components are represented
```

Comment:
You may want to say "in our current NeoPixels". Someday we might buy fancy ones. You could leave a TODO for when we upgrade? Or state that this code is not intended to be compatible with 4-component NeoPixels (only 3 color components)